agent_build/common.py

```diff
@@ -0,0 +1,341 @@
+# Copyright 2014-2021 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import pathlib as pl
+import re
+import shlex
+import subprocess
+import sys
+import time
+from typing import Union, List
+
+__SOURCE_ROOT__ = pl.Path(__file__).parent.parent
+
+
+def cat_files(file_paths, destination, convert_newlines=False):
+    """Concatenates the contents of the specified files and writes it to a new file at destination.
+
+    @param file_paths: A list of paths for the files that should be read. The concatenating will be done in the same
+        order as the list.
+    @param destination: The path of the file to write the contents to.
+    @param convert_newlines: If True, the final file will use Windows newlines (i.e., CR LF).
+    """
+    dest_fp = open(destination, "w")
```

Comment:
Ideally should use ``with`` context manager or ``try / finally`` to ensure those file handles are always closed.

Reply:
Yeah, I've just copied almost everything in this file directly from the existing `build_package.py` file, so it may contain some things to change. Will do.